packages/components/modal/__specs__/modal.spec.tsx

```diff
@@ -32,4 +33,18 @@ describe("<Modal />", () => {
 
     expect(modal.baseElement).toMatchSnapshot();
   });
+
+  context("x button", () => {
+    it("closes the modal", async () => {
+      const { user, getByTestId } = render(<MockModal />);
+
+      act(() => {
+        user.click(getByTestId("close-modal-button"));
```

Comment:
If we use the `aria-role` attribute we can potentially move this to use `getByRole` which is the recommended selector in the priority list of `@testing-library/react` - https://testing-library.com/docs/queries/about#priority

Reply:
@dgalarza Just adding `role="button"` is not sufficient enough to find the X button (it returns all the buttons in the modal). The most semantic way I could think to target the X button based on the documentation was adding a hidden `Body` element:

component: 
```
<Box
    aria-describedby="x-button"
    height="26px"
    onClick={closeModal}
    position="relative"
    role="button"
    style={{ cursor: "pointer" }}
    width="26px"
  >
    <Box
      bg="grey50"
      height="1.5px"
      left="5.5px"
      position="absolute"
      style={{ rotate: "45deg" }}
            top="50%"
      width="15px"
    />
    <Box
      bg="grey50"
      height="1.5px"
      left="5.5px"
      position="absolute"
      style={{ rotate: "135deg" }}
      top="50%"
      width="15px"
    />
    <Body display="none" id="x-button" size="s">X</Body>
</Box>
  ```
  
  spec:
  ```user.click(getByRole("button", { description: "X" }));```

Reply:
@wmurray would `aria-label` work as well?

Reply:
I was thinking:

`user.click(getByRole("button", { description: "Close Button" }));` where "Close Button" is the aria-label content.

> getByRole: This can be used to query every element that is exposed in the [accessibility tree](https://developer.mozilla.org/en-US/docs/Glossary/AOM). With the name option you can filter the returned elements by their [accessible name](https://www.w3.org/TR/accname-1.1/).


> 5. Accessible Name and Description Mapping
Information concerning name and description accessibility API mappings, including relationships, such as labelled-by/label-for and described-by/description-for, is documented in the [Core Accessibility API Mappings](https://www.w3.org/TR/core-aam-1.1/) specification [[CORE-AAM-1.1](https://www.w3.org/TR/accname-1.1/#bib-core-aam-1.1)]. See the mapping table entries for [aria-label](https://www.w3.org/TR/core-aam-1.1/#ariaLabel), [aria-labelledby](https://www.w3.org/TR/core-aam-1.1/#ariaLabelledBy), and [aria-describedby](https://www.w3.org/TR/core-aam-1.1/#ariaDescribedBy).
https://www.w3.org/TR/accname-1.1/#accessible-name-and-description-mapping

Reply:
Got it!